lib/Runtime/Base/FunctionInfo.h

```diff
@@ -78,6 +79,8 @@ namespace Js
         static bool IsCoroutine(Attributes attributes) { return ((attributes & (Async | Generator)) != 0); }
         bool IsCoroutine() const { return IsCoroutine(this->attributes); }
         bool HasComputedName() const { return (this->attributes & Attributes::ComputedName) != 0; }
+        bool HasCachedScope() const { return (this->attributes & Attributes::CachedScope) != 0; }
```

Comment:
A naming question: is this attribute recording the fact that a function has a cached scope or that it needs a scope object and thus *may* have a cached scope?